src/layer.js

```diff
@@ -234,6 +234,34 @@ export class MapLayer extends HTMLElement {
       }
     }, 0);
   }
+  getOuterHTML() {
+    let fullLayer = this.outerHTML;
+    let tempElement = document.createElement('div');
+    tempElement.innerHTML = fullLayer;
+
+    if (this.hasAttribute('src')) {
+      let newSrc = this._layer.getHref();
+      tempElement.querySelector('layer-').setAttribute('src',newSrc);
+    }
+    if (this.querySelector('map-link')) {
+      let mapLinks = tempElement.querySelectorAll('map-link');
+
+      mapLinks.forEach((mapLink) => {
+       
+        if (mapLink.hasAttribute('href')) {
+          mapLink.setAttribute('href', decodeURI(this.transformURL(mapLink.attributes.href.value, this.baseURI ? this.baseURI : document.baseURI)));
+        }
+        else if (mapLink.hasAttribute('tref')) {
+          mapLink.setAttribute('tref', decodeURI(this.transformURL(mapLink.attributes.tref.value, this.baseURI ? this.baseURI : document.baseURI)));
```

Comment:
I wasn't certain that `new URL(...)` would work with a `tref` value as input, due to the presence of `{..}` 

How sure are we that the base we're resolving against is correct? i.e. what is the difference between `this.baseURI` and `document.baseURI`?

Reply:
I think the document.baseURL is just a last case scenario like mentioned in _ready()